### DISCUSSION TITLE
Morphir, Ditto and FINOS Content Collaboration

### DISCUSSION THREAD

**Discussion by ZnglyJess** [finos/morphir#76] [General]
This discussion is to being the Morphir, Ditto and FINOS teams together around the content work that's happening for the FINOS Morphir project.

**Comment by zngly-vlad** [comment]
Hello everyone, here is the live GitHub pages link to the morphir docusaurus website mockup by the Zngly team. 
I would love to get everyone's feedback in this discussion thread. 

https://zngly.github.io/morphir/

@stephengoldbaum @mcleo-d @AttilaMihaly

**Reply by mcleo-d** [reply]
Hi @ditto-vlad - Thanks for providing the link for feedback. I have raised the following PR to start migrating existing content into the new microsite design. We can then work with @stephengoldbaum to review and deprecate unwanted content as part of the review.

https://github.com/zngly/morphir/pull/11

**Reply by zngly-vlad** [reply]
@mcleo-d The pr has been merged. Some actions have been taken in order to deploy.

Many links were broken so the default `onBrokenLinks: 'throw'` has been changed to `onBrokenLinks: 'warn'` in `docusaurus.config.js`

https://zngly.github.io/morphir/docs/intro

**Reply by mcleo-d** [reply]
Thanks @ditto-vlad - The content contributed in https://github.com/zngly/morphir/pull/11 can now be disregarded in favour of reviewing the content and information architecture in the upstream `morphir` repo after https://github.com/finos/morphir/pull/78 is merged.

**Reply by mcleo-d** [reply]
Hi @ditto-vlad - Can this discussion be closed now the `zngly morphir` repo has been deprecated?

**Comment by ZnglyJess** [comment]
@stephengoldbaum please find the latest version of the morphir factsheet copy, we have incorporated the edits requested by @mcleo-d 

Ive attached the old PDF version for layout ref too: [Morphir_Factsheet.pdf](https://github.com/finos/morphir/files/8032684/Morphir_Factsheet.pdf)

# Welcome to Morphir
## A library of tools that works to capture business logic as data.

## For the first time, business logic can be shared, stored, translated and visualised, all with the reliability of standardisation ensured in the Morphir framework.

### What is it?
A set of tools for integrating technologies. Morphir is composed of a library of tools that facilitate the digitisation of business logic into multiple different languages & platforms. The Morphir framework is unique too in that it facilitates elements of automation and conversion that were previously unavailable in the field of fintech.

[we need a diagram or graphic to go here]

### Why is it important?
Morphir makes business logic portable. Business logic digitised provides distinct advantages: capacity for movement across departments and fields & the ability to be converted to new languages and applications.

It allows for use of existing and future tools. Morphir ensures that business logic conforms to strict standards, future-proofing new languages and application modelling adaptations as well as current.

### How does it work?
Defines a standard format for storing and sharing business logic. A clear set of standards and format is in-place from the input/output, allowing for coherent structure. 

Provides APIs and tools to integrate technologies. Once your business logic is converted into developer-friendly data, ease of integration and assimilation into new environments is possible. 

## Morphir is a library of tools that works to capture business logic as data.

### What are the benefits?


#### Eliminates technical debt risk  
Refactoring code libraries is often a harmful and time-sensitive issue for businesses, Morphir ensures the standards introduced from input eliminate delays at deployment.

#### Ensures correctness
Certifying that specified functions behave as intended from input to output is assured through the Morphir library / tool chain.

#### Increases agility
Adaptability and usability are key concepts of the Morphir framework, business logic can now move with the code, be easily understood and adopted, in an ever-developing ecosystem. 

#### Disseminates information through automation
Morphir’s automated processing helps disseminate information which otherwise may not be understood or shared at all, a useful tool when bringing elements of business logic to conversation outside of its immediate audience (i.e developers).

----

### An ecosystem of innovative features.
Supporting the development of your business’ needs in an ever-developing ecosystem based on firm standards and  the integration of new languages.

Button - [link to github]

----

#### FINOS: We're a community creating open-source solutions for Financial Services
Providing an independent setting to deliver software and standards that address common industry challenges and drive innovation.

**Reply by mcleo-d** [reply]
Hi @ZnglyJess - Thanks for sharing the content on GitHub as this will make everything easy for @stephengoldbaum and @AttilaMihaly to review.

I would like to question the use of the H2 format below as the current `##` are statements rather than headers?

```
# Welcome to Morphir
## A library of tools that works to capture business logic as data.

## For the first time, business logic can be shared, stored, translated and visualised, all with the reliability of standardisation ensured in the Morphir framework.
```

Should the format not be **bold** as the following ...?

```
# Welcome to Morphir
**A library of tools that works to capture business logic as data.**

**For the first time, business logic can be shared, stored, translated and visualised, all with the reliability of standardisation ensured in the Morphir framework.**
```

It would be great to get your thoughts 😸

**Reply by ZnglyJess** [reply]
Hi @mcleo-d - yes ideally when its set in pdf format there will be better differentiation between the main H1 and the other H2 areas - this is probably better demonstrated in the pdf i attached. See here: [Morphir_Factsheet (1).pdf](https://github.com/finos/morphir/files/8032813/Morphir_Factsheet.1.pdf)

**Reply by mcleo-d** [reply]
Thanks @ZnglyJess - Trust a techie to be looking at the tagging rather than the content!

**Reply by mcleo-d** [reply]
Hi @ZnglyJess - Can this discussion be closed now the `zngly morphir` repo has been deprecated?

**Reply by ZnglyJess** [reply]
> Hi @ZnglyJess - Can this discussion be closed now the `zngly morphir` repo has been deprecated?

how do i do that @mcleo-d ?

**Comment by zngly-vlad** [comment]
Hello EveryBody, 

I have to created a new [PR](https://github.com/finos/morphir/pull/82) to address the resource center destination url.

I would like to pivot this discussion to request a demo from one of the developers at morphir. It would greatly help getting to know the product and moving the new website build documentation into the right direction, coming up with a good format to help the developers really get excited about the morphir project.

@mcleo-d @stephengoldbaum @AttilaMihaly

**Reply by mcleo-d** [reply]
Hi @ditto-vlad

Great idea about the information architecture work. This falls in line with @colineberhardt's recommendation for looking at the information architecture after the microsite build had been merged cc @stephengoldbaum.

Right now, the following PRs are open. I recommend these are merged before we have a microsite content review.

- #82
- #83
- #84

James.

**Reply by mcleo-d** [reply]
FYI - The deployed `morphir` `docusaurus` microsite can be found here - https://morphir.finos.org/

**Comment by stephengoldbaum** [comment]
Good day.  I've approved and merged all of the latest (I think).  Looks great!!!

@ditto-vlad @mcleo-d

**Reply by psmulovics** [reply]
Not just great - amazing! Totally 21st century feel! :)

**Comment by stephengoldbaum** [comment]
Vlad, I'm happy to give you a demo if you want to send over some times.

**Reply by zngly-vlad** [reply]
Hello Stephen, 

My apologies, haven't seen this in proper time.

I would love spend some get a demo when you get a chance. I am free most workdays regarding meetings, so whenever suits you is best.

In the meantime I created a repo where I messed around with morphir demos, it is super simple to install and get it running on your local machine. It is essentially a starter-app, which would give developers quick preview of how morphir works.

My vision is a npx command somewhere along the lines of: `npx @morphir/starter`. But I am just messing around for the moment.

https://github.com/zngly/morphir-example

**Comment by zngly-vlad** [comment]
Hello everybody,

I have given the morphir docs a thought regarding the structure of the content and would like to propose the idea. 

Following in line with practices I have seen and like, It is important that the reader get a chance to know what morphir is, does, and how easy it is to dive in. Nothing is more import and to a dev than the strong documentation that a tool provides. After some brainstorming I have come up with the following:

--- Start Morphir Docs SideBar Structure --- 

- ### Introduction
  - Intro
  - Background

- ### Set-Up
  - Installation
  - Quick Tutorial

- ### API Reference
  - Documentation regarding morphir...?
  - Not Sure exactly what these are for morpir but e.g.
  - #### ELM
    - morphir-elm dev
    - morphir-elm make
    - morphir-elm gen

- ### Morphir Concepts
  - #### Architecture
    - Why functional Programming
  - #### Modelling
    - Financial Concepts
    - Modelling for Devs

- ### Resources
  - Tutorials by stephen
  - morphir factsheet
  - finos resource centre
 
- ### Community
  - slack
  - github
  - contrbuting

--- End Morphir Docs SideBar Structure

I think this would be a good starter design to give structure to the morphir docs. I would love to hear everybodys opinion and get feedback on how we can improve this. 

@mcleo-d @AttilaMihaly  @stephengoldbaum @ZnglyJess

**Reply by stephengoldbaum** [reply]
Looks great and we've been wanting to structure documents for a long time.  I think after the Setup section could be a Using section.  We could actually split that for different target audiences.  We have the modelers writing the business logic, the developers writing integration tools to add features for their situations, and contributors (which would probably have overlap with the developers).
